scripts/yocto-build-and-test.sh

```diff
@@ -148,9 +148,22 @@ prepare_build_config() {
 
     local mender_binary_delta_version=$($WORKSPACE/mender-binary-delta/x86_64/mender-binary-delta --version | egrep -o '[0-9]+\.[0-9]+\.[0-9b]+(-build[0-9]+)?')
     cat >> $BUILDDIR/conf/local.conf <<EOF
-FILESEXTRAPATHS${sep}prepend${sep}pn-mender-binary-delta := "$WORKSPACE/mender-binary-delta:"
 PREFERRED_VERSION${sep}pn-mender-binary-delta = "$mender_binary_delta_version"
 EOF
+    # This condition is not actually related to the separator, but the new way
+    # to include mender-binary-delta binaries happened to change at the same
+    # time as the seperator, in kirkstone. Once that branch is dropped, we can
+    # keep only the SRC_URI variant below.
+    if [ "$sep" = "_" ]; then
+        cat >> $BUILDDIR/conf/local.conf <<EOF
+FILESEXTRAPATHS${sep}prepend${sep}pn-mender-binary-delta := "$WORKSPACE/mender-binary-delta:"
+EOF
+    else
+        local tarball=$(echo $WORKSPACE/mender-binary-delta/*.tar.xz)
+        cat >> $BUILDDIR/conf/local.conf <<EOF
+SRC_URI${sep}pn-mender-binary-delta = "file://${tarball}"
+EOF
+    fi
```

Comment:
alternative solution:
```
local tarballs=("$WORKSPACE"/mender-binary-delta/*.tar.xz)
        cat >> "$BUILDDIR"/conf/local.conf <<EOF
SRC_URI${sep}pn-mender-binary-delta = "file://${tarballs[0]}"
EOF
```

which is clear, saves extra shell execution, and underlines the fact that we are taking the first tar ball. this code does not asserts that there is only one.

Reply:
Yes, but it should fail if that happens. Actually my subshell approach also succeeds, but it results in an invalid `SRC_URI`, which will fail later once Bitbake starts scanning the `SRC_URI`.

OTOH, yours will silently succeed. I agree that this would have been handy in other situations, but not here.

Reply:
ok, that goes over my explicit objection. at least, please add double quotes around $WOKRSPACE and $BUILDIR.

Reply:
Done!